### PR TITLE
🐛 aws secrets manager and security groups resource fixes

### PR DESCRIPTION
**Pull request by vjeffrey**
fixes https://github.com/mondoohq/cnquery/issues/1468
fixes https://github.com/mondoohq/cnquery/issues/1467

### CODE REVIEW

**Comment by czunker** [comment]
I gave it a try, and it gets populated, but something is wrong with the actual values.

This is the output from cnquery:
```
cnquery run aws -c "aws.ec2.securityGroups.where(id == /sg-019e35122732c2547/){ ipPermissions {*} ipPermissionsEgress {*} name }" 
...
aws.ec2.securityGroups.where: [
  0: {
    name: "launch-wizard-28"
    ipPermissionsEgress: [
      0: {
        ipRanges: [
          0: "0.0.0.0/0"
        ]
        toPort: 22
        id: "sg-019e35122732c2547-0"
        fromPort: 22
        ipv6Ranges: []
        ipProtocol: "tcp"
      }
    ]
    ipPermissions: [
      0: {
        ipRanges: [
          0: "0.0.0.0/0"
        ]
        ipProtocol: "tcp"
        toPort: 22
        id: "sg-019e35122732c2547-0"
        ipv6Ranges: []
        fromPort: 22
      }
    ]
  }
]
```

`ipPermissions` and `ipPermissionsEgress` are the same.
The output from the was CLI looks different:
```
aws ec2 describe-security-group-rules --region us-east-2 --filter Name=group-id,Values=sg-019e35122732c2547 | cat                                                                    ✔ │ 08:20:02 
{
    "SecurityGroupRules": [
        {
            "SecurityGroupRuleId": "sgr-0b651d6dd98e469b3",
            "GroupId": "sg-019e35122732c2547",
            "GroupOwnerId": "921877552404",
            "IsEgress": true,
            "IpProtocol": "-1",
            "FromPort": -1,
            "ToPort": -1,
            "CidrIpv4": "0.0.0.0/0",
            "Tags": []
        },
        {
            "SecurityGroupRuleId": "sgr-0ee256f8f56f2ccad",
            "GroupId": "sg-019e35122732c2547",
            "GroupOwnerId": "921877552404",
            "IsEgress": false,
            "IpProtocol": "tcp",
            "FromPort": 22,
            "ToPort": 22,
            "CidrIpv4": "0.0.0.0/0",
            "Tags": []
        }
    ]
}

```

**Comment by vjeffrey** [comment]
oooh, i bet it's an id conflict within mql, ill take a look, thanks for checking!

**Comment by czunker** [comment]
> oooh, i bet it's an id conflict within mql, ill take a look, thanks for checking!

Yes, you are correct. The ids are the same:
```
aws.ec2.securityGroups.where: [
  0: {
    name: "launch-wizard-28"
    ipPermissions: [
      0: {
        id: "sg-019e35122732c2547-0"
      }
    ]
    ipPermissionsEgress: [
      0: {
        id: "sg-019e35122732c2547-0"
      }
    ]
  }
]
```

File: resources/packs/aws/aws_ec2.go [hunk 1]
```diff
@@ -301,6 +301,40 @@ func (s *mqlAwsEc2) getSecurityGroups(provider *aws_provider.Provider) []*jobpoo
...
+
+						ipv6Ranges := []interface{}{}
+						for r := range permission.Ipv6Ranges {
+							iprange := permission.Ipv6Ranges[r]
+							if iprange.CidrIpv6 != nil {
+								ipRanges = append(ipRanges, *iprange.CidrIpv6)
+							}
+						}
+						mqlSecurityGroupIpPermission, err := s.MotorRuntime.CreateResource("aws.ec2.securitygroup.ippermission",
+							"id", core.ToString(group.GroupId)+"-"+strconv.Itoa(p),
```
Comment (czunker) [inline resources/packs/aws/aws_ec2.go]:
```suggestion
							"id", core.ToString(group.GroupId)+"-"+strconv.Itoa(p)+"-egress",
```

I had a similar problem with the networkAcls: https://github.com/mondoohq/cnquery/pull/1508

Comment (vjeffrey) [reply resources/packs/aws/aws_ec2.go]: oh thank you!

**Review by czunker — APPROVED** [review]
Looking good now:
```
aws.ec2.securityGroups.where: [
  0: {
    name: "launch-wizard-28"
    ipPermissionsEgress: [
      0: {
        ipRanges: [
          0: "0.0.0.0/0"
        ]
        toPort: 0
        id: "sg-019e35122732c2547-0-egress"
        fromPort: 0
        ipv6Ranges: []
        ipProtocol: "-1"
      }
    ]
    ipPermissions: [
      0: {
        ipRanges: [
          0: "0.0.0.0/0"
        ]
        ipProtocol: "tcp"
        toPort: 22
        id: "sg-019e35122732c2547-0"
        ipv6Ranges: []
        fromPort: 22
      }
    ]
  }
]
```

**Pull request merged**